cloud/scope/machine.go

```diff
@@ -544,12 +544,12 @@ func (m *MachineScope) ReconcileCreateInstanceOnLB(ctx context.Context) error {
 			return err
 		}
 		backendSet := lb.BackendSets[APIServerLBBackendSetName]
-		if !m.containsBackend(backendSet, m.Name()) {
+		if !m.containsNLBBackend(backendSet, m.Name()) {
```

Comment:
can we add test to machine_test.go please? test that when cluster spec is set to lb, then the backend is added to lbaas.

Reply:
Added test cases for Load Balancer in latest commit.